app/models/shipment.rb

```diff
@@ -6,15 +6,12 @@ class Shipment < ApplicationRecord
 
   def items_to_array
     #每個 shipment 物品清單
-    items = shipment_items.map { |item| item.description}
-    items_uniq = items.uniq
-
-    counts = items_uniq.map{ |item| items.count(item) }
-   
+    arr = shipment_items.group("description").count.sort_by{ |_k, v|  v }.reverse
```

Comment:
Can reduce to
```suggestion
    arr = shipment_items.group("description").count.sort_by{ |_k, v|  -v }
```
But there is another better solution order in single query, I'll leave it to you 😬

Reply:
OK, I'll search as I could!